### PR TITLE
Export Task Providers to a separate class

### PR DESCRIPTION
**Pull request by cortinico**
Moved all the task providers to a separate file and added test coverage for them.
Plus I've updated all the parameters from `ApplicationVariant` to `String` so become trivial to test.

### CODE REVIEW

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     * It could be either ProGuard or R8
+     *
+     * @return the task or null otherwise
+     */
+    @JvmStatic
+    fun getTransformerTask(project: Project, variantName: String): Task? =
+        project.findTask(
+            // Android Studio 3.3 includes the R8 shrinker.
+            "transformClassesAndResourcesWithR8For${variantName.capitalized}",
+            "transformClassesAndResourcesWithProguardFor${variantName.capitalized}",
```
Comment (marandaneto) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt:20]: I wonder if this is still necessary if we support AGP 4.0 and up only

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+     *
+     * @return the task if found or null otherwise
+     */
+    @JvmStatic
+    fun getAssembleTask(project: Project, variant: ApplicationVariant): Task {
+        return try {
+            variant.assembleProvider.get()
+        } catch (ignored: Exception) {
+            project.logger.error("[sentry] .assembleProvider failed with: ${ignored.message}")
+            variant.assemble
```
Comment (marandaneto) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt:80]: I wonder if this is still necessary if we support AGP 4.0 and up only

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryTaskProviderTest.kt [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    fun `getTransformerTask returns null for missing task`() {
+        val project = ProjectBuilder.builder().build()
+
+        assertNull(getTransformerTask(project, "debug"))
+    }
+
+    @Test
+    fun `getTransformerTask returns transform for R8`() {
+        val project = ProjectBuilder.builder().build()
+        val task = project.tasks.register("transformClassesAndResourcesWithR8ForDebug")
```
Comment (marandaneto) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryTaskProviderTest.kt:32]: these 2 lines are repeated across all the tests, maybe let's add a method to not DRY?

`registerTask(name: String): Tuple<Project, Task>` or something like that

what do you think?

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt (hunk 1 above)
Comment (cortinico) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt:20]: True, thought the CI matrix runs on AGP 3.4 and above so is probably worth to keep.

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt (hunk 2 above)
Comment (cortinico) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryTasksProvider.kt:80]: True, thought the CI matrix runs on AGP 3.4 and above so is probably worth to keep.

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryTaskProviderTest.kt (hunk 3 above)
Comment (cortinico) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryTaskProviderTest.kt]: Yup added 👍

**Review by marandaneto — APPROVED** [review]
LGTM

**Pull request merged**